samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -201,7 +202,7 @@ private String createProcessorId(Config config) {
     ApplicationConfig appConfig = new ApplicationConfig(config);
     if (appConfig.getProcessorId() != null) {
       return appConfig.getProcessorId();
-    } else if (appConfig.getAppProcessorIdGeneratorClass() != null) {
+    } else if (!StringUtils.isEmpty(appConfig.getAppProcessorIdGeneratorClass())) {
```

Comment:
Can we replace `isEmpty` w/ `isBlank` so that values like `"   "` are also not accepted?
btw, commons does have `isNotEmpty`, `isNotBlank`. IMO, it makes it more readable and explicit.